source/agora/consensus/protocol/Nominator.d

```diff
@@ -891,14 +891,17 @@ extern(D):
 
         const pendingBlockSig = (block_sig.height == this.pending_block.header.height);
 
-        if (!pendingBlockSig && block_sig.height > cur_height) // Too early for this signature
+        Block block;
+        if (pendingBlockSig)
+            block = this.pending_block;
+        else if (block_sig.height <= cur_height)
+            block = this.ledger.getBlocksFrom(Height(block_sig.height)).front;
+        else // Too early for this signature
         {
             early_sigs.put(block_sig);
-            return BlockHeader.init; // We want the caller to ignore this signature
```

Comment:
Why remove the comment. It is still valid.

Reply:
I missed it carelessly. Will restore it.

Reply:
Updated.